app/src/main/java/com/boolder/boolder/view/map/MapFragment.kt

```diff
@@ -118,6 +118,7 @@ class MapFragment : Fragment(), BoolderMapListener {
                     when (newState) {
                         STATE_EXPANDED -> mapViewModel.onProblemTopoVisibilityChanged(isVisible = true)
                         STATE_HIDDEN -> {
+                            mapViewModel.onTopoUnselected()
```

Comment:
Looks like a call to `mapView.unselectProblem()` is missing, as the boulder problem stays selected when dismissing the topo bottom sheet from a back press or dragging it down (the `unselectProblem()` function's scope can be changed from private to public)